.github/workflows/pixi.toml

```diff
@@ -6,53 +6,123 @@ channels = ["https://prefix.dev/conda-forge"]
+solve-group = "array-api-cpu"
+
+[environments.jax-cpu]
+# tasks: test-jax
+features = ["build-deps", "test-deps", "mkl", "jax-cpu", "jax-cpu-tasks"]
+solve-group = "array-api-cpu"
+
+[environments.torch-cpu]
+# tasks: test-torch, test-torch-float32
+features = ["build-deps", "test-deps", "mkl", "torch-cpu", "torch-cpu-tasks"]
+solve-group = "array-api-cpu"
+
+[environments.array-api-cpu]
+# tasks: test-cpu
+features = ["build-deps", "test-deps", "test-cpu", "mkl", "array_api_strict", "dask", "jax-cpu", "torch-cpu"]
+solve-group = "array-api-cpu"
+
+[environments.build-cuda]
+# tasks: build-cuda
+# note: py-cuda feature ensures compatibility with both cuda12 and cuda13 solve groups
+features = ["build-deps", "build-cuda", "py-cuda"]
+
+[environments.cupy]
+# tasks: test-cupy
+features = ["build-deps", "test-deps", "cuda13", "py-cuda", "cupy"]
+solve-group = "cuda13"
+
+[environments.jax-cuda]
+# tasks: test-jax-cuda
+features = ["build-deps", "test-deps", "cuda12", "py-cuda", "mkl", "jax-cuda"]
+solve-group = "cuda12"
+
+[environments.torch-cuda]
+# tasks: test-torch-cuda
+features = ["build-deps", "test-deps", "cuda12", "py-cuda", "mkl", "torch-cuda"]
+solve-group = "cuda12"
+
+
+### Default dependencies (included in all environments) ###
+
 [dependencies]
-spin = ">=0.14,<0.15"
-python = ">=3.13.7,<3.14"
-
-
-### Basic building ###
-
-[feature.build.dependencies]
-compilers = ">=1.11.0,<2"
-pkg-config = ">=0.29.2,<0.30"
-ninja = ">=1.13.1,<2"
-meson = ">=1.9.0,<2"
-meson-python = ">=0.18.0,<0.19"
-cython = ">=3.1.4,<4"
-python-build = ">=1.3.0,<2"
-pip = ">=25.2,<26"
-blas-devel = ">=3.9.0,<4"
-numpy = ">=2.3.3,<3"
-pybind11 = ">=3.0.1,<4"
-pythran = ">=0.18.0,<0.19"
-
-[feature.build.tasks.build]
+spin = "*"
+python = "*"
+
+
+### Default building ###
+
+[feature.build-deps.dependencies]
+compilers = "*"
+pkg-config = "*"
+ninja = "*"
+meson = "*"
+meson-python = "*"
+cython = "*"
+blas-devel = "*"
+numpy = "*"
+pybind11 = "*"
+pythran = "*"
+
+# XXX: when updating this task, remember to update other build tasks if appropriate
+[feature.build-task.tasks.build]
 cmd = "spin build --setup-args=-Dblas=blas --setup-args=-Dlapack=lapack --setup-args=-Duse-g77-abi=true"
 cwd = "../.."
 env = { CC = "ccache $CC", CXX = "ccache $CXX", FC = "ccache $FC" }
 description = "Build SciPy (default settings)"
 
 
-### Basic testing ###
+### Default testing ###
 
-[feature.test.dependencies]
-pytest = ">=8.4.2,<9"
-hypothesis = ">=6.139.2,<7"
-pytest-cov = ">=7.0.0,<8"
-pytest-timeout = ">=2.4.0,<3"
-pytest-xdist = ">=3.8.0,<4"
-threadpoolctl = ">=3.6.0,<4"
-pooch = ">=1.8.2,<2"
-mpmath = ">=1.3.0,<2"
-gmpy2 = ">=2.2.1,<3"
-ccache = ">=4.11.3,<5"
-marray-python = ">=0.0.12,<0.0.13"
+[feature.test-deps.dependencies]
+pytest = "*"
+hypothesis = "*"
+pytest-cov = "*"
+pytest-timeout = "*"
+pytest-xdist = "*"
+threadpoolctl = "*"
+pooch = "*"
+mpmath = "*"
+gmpy2 = "*"
+ccache = "*"
+marray-python = "*"
```

Comment:
`marray` is an array API dependency only, right? If so, it should be moved and treated similarly to, e.g., `dask`.